kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/filter/NettyFilterDispatchExecutor.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.filter;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.netty.channel.EventLoop;
+
+import io.kroxylicious.proxy.filter.FilterDispatchExecutor;
+
+public class NettyFilterDispatchExecutor implements FilterDispatchExecutor {
+
+    private final EventLoop eventLoop;
+
+    public NettyFilterDispatchExecutor(EventLoop eventLoop) {
+        Objects.requireNonNull(eventLoop, "eventLoop cannot be null");
+        this.eventLoop = eventLoop;
+    }
+
+    @Override
+    public boolean isInFilterDispatchThread() {
+        return eventLoop.inEventLoop();
+    }
+
+    public EventLoop getEventLoop() {
```

Comment:
do we really need this on the public API?

Reply:
nope, I wanted it for testing (should add a comment). but this whole implementation class in general shouldn't be so publicly available. I'll see if I can make it package private.